concert/app/src/functions/on_event_stage_changed/index.py

```diff
@@ -1,32 +1,54 @@
 import boto3
 import json
 import os
+import decimal
+from boto3.dynamodb import types
 from enchanted_brain.attributes import (
+    ATTR_AGGREGATE_CHOICE_COUNT,
+    ATTR_AGGREGATE_CHOICE_SUM,
+    ATTR_CHOICE_TIMESTAMP,
+    ATTR_CHOICE_TYPE,
+    ATTR_CHOICE_VALUE_CHILLS,
+    ATTR_CHOICE_VALUE_COLOR,
     ATTR_EVENT_STAGE_ID,
     ATTR_RECORD_ID,
+    ATTR_SONG_LIST_CHOICES,
     ATTR_SONG_LIST_DISPLAY_NAME,
     ATTR_SONG_LIST_END_TIME,
     ATTR_SONG_LIST_SONGS,
     ATTR_SONG_LIST_START_TIME,
+    EVENT_STAGE_END,
+    RECORD_ID_AGGREGATE,
     RECORD_ID_EVENT_STAGE,
     RECORD_ID_SONG_LIST,
 )
+from enchanted_brain.parser import DynamoDbEncoder
 
 CALLBACK_GLOBAL_SNS_TOPIC_ARN = os.environ["CALLBACK_GLOBAL_SNS_TOPIC_ARN"]
 DYNAMODB_TABLE_NAME = os.environ.get("DYNAMODB_TABLE_NAME")
 
 sns = boto3.client("sns")
 dynamodb = boto3.client("dynamodb")
 
+dynamodb_encoder = DynamoDbEncoder()
+dynamodb_serializer = boto3.dynamodb.types.TypeSerializer()
+dynamodb_deserializer = boto3.dynamodb.types.TypeDeserializer()
+
 
 def handler(event, context):
     message = json.loads(event["body"])
 
+    if message["data"][ATTR_EVENT_STAGE_ID] == EVENT_STAGE_END:
+        song_list, aggregate_data = get_song_list_and_aggregate_data()
+        message["data"][ATTR_SONG_LIST_SONGS] = get_songs_with_aggregate_choices(
+            song_list, aggregate_data
+        )
+
     update_event_stage_and_song_list(message)
 
     sns.publish(
         TopicArn=CALLBACK_GLOBAL_SNS_TOPIC_ARN,
-        Message=json.dumps(message),
+        Message=json.dumps(message, default=dynamodb_encoder.default),
```

Comment:
In OnConnectFunction, it's `json.dumps(message, cls=DynamoDbEncoder)`. Do they both work?